File: topics/divide-and-conquer-algorithms/java/BSIter.java
public class BSIter {
    public static void main(String[] args) {
        int[] array = {3,4,5,6,7,8,9};
        int pos = binarySearch(array, 4, 0, array.length-1);
        if(pos == -1) {
            System.out.println("Not found");
        } else {
            System.out.println("Found at index " + pos);
        }
    }

    static int binarySearch(int array[], int x, int low, int high) {
        while(high >= low) {
            int mid = (high+low)/2;
            if(x == array[mid]) {
                return mid;
            }
            else if (x > array[mid]) {
                low = mid + 1;
            }
            else {
                high = mid -1;
            }
        }
        return -1;
    }
}

File: topics/divide-and-conquer-algorithms/python/BSIter.py
def binarySearch(arr:list, x:int, low:int, high:int)->int:
    while high>=low:
        mid = int((high+low)/2)
        if x == arr[mid]:
            return mid
        elif x > arr[mid]:
            low = mid + 1
        else:
            high = mid -1
    return -1

if __name__== "__main__":
    arr: list = [3,4,5,6,7,8,9]
    pos = binarySearch(arr, 4, 0, len(arr))
    if pos == -1:
        print('Element not found')
    else:
        print('Element found at ' + str(pos))